packages/dashboard/src/customization/widgets/text/component.test.tsx

```diff
@@ -13,88 +11,101 @@ jest.mock('~/store/actions', () => ({
-    { readOnly: false, isSelected: false, isUrl: true },
-    { readOnly: false, isSelected: true, isUrl: true },
-    { readOnly: false, isSelected: true, isUrl: false },
-    { readOnly: true, isSelected: false, isUrl: true },
-  ].forEach((configuration) => {
-    it(`should render ${JSON.stringify(configuration)} correctly`, () => {
-      const { isSelected, readOnly, isUrl } = configuration;
-      (useIsSelected as jest.Mock).mockImplementation(() => isSelected);
-      (useSelector as jest.Mock).mockImplementation(() => readOnly);
-      (useDispatch as jest.Mock).mockReturnValue(jest.fn());
-
-      const { container } = render(
-        <TextWidgetComponent
-          id='some-id'
-          x={1}
-          y={2}
-          z={3}
-          height={100}
-          width={100}
-          type='text-widget'
-          properties={{ isUrl, value: 'abc' }}
-        />
-      );
-
-      expect(container).toMatchSnapshot();
-    });
+  it('should render editable text', () => {
+    const text = 'text';
+    render(<TestEditableTextWidget value={text} />);
+
+    const textWidget = screen.getByText(text);
+    expect(textWidget).toBeVisible();
+    expect(textWidget).not.toHaveAttribute('aria-readonly');
+    expect(screen.queryByRole('link', { name: text })).not.toBeInTheDocument();
   });
 
-  it('should exit edit mode when unmounted', () => {
-    (useDispatch as jest.Mock).mockImplementation(() => jest.fn((cb) => cb())); // short curcuit dispatch
-
-    (useIsSelected as jest.Mock).mockImplementation(() => false);
-    (useSelector as jest.Mock).mockImplementation(() => false);
-    (useDispatch as jest.Mock).mockReturnValue(jest.fn());
-
-    const { unmount } = render(
-      <TextWidgetComponent
-        id='some-id'
-        x={1}
-        y={2}
-        z={3}
-        height={100}
-        width={100}
-        type='text-widget'
-        properties={{ isUrl: false, value: 'abc' }}
-      />
-    );
-    unmount();
-
-    expect(onChangeDashboardGridEnabledAction).toBeCalledWith({
-      enabled: true,
-    });
+  it('should render editable link', () => {
+    const text = 'text';
+    const href = 'https://test.com';
+    render(<TestEditableTextWidget value={text} href={href} isUrl />);
+
+    const textWidget = screen.getByText(text);
+    expect(textWidget).toHaveTextContent(text);
+    expect(textWidget).not.toHaveAttribute('aria-readonly');
+    expect(screen.queryByRole('link', { name: text })).not.toBeInTheDocument();
+  });
+
+  it('should santize and then render unsafe link', () => {
+    const text = 'text';
+    const href = 'javascript://%0Aalert(1)';
+    render(<TestReadonlyTextWidget value={text} href={href} isUrl />);
+
+    const textWidget = screen.getByText(text);
+    expect(textWidget).toBeVisible();
+    expect(textWidget).not.toHaveAttribute('href');
+    expect(screen.queryByRole('link', { name: text })).not.toBeInTheDocument();
+  });
+
+  it('should santize and then render unsafe link 2', () => {
```

Comment:
[Nit]: It is hard to tell what specifically the tests are testing for beyond "not unsafe" due to the duplication of the name and the usage of a number (2) to distinguish. It would be better if the test names were more explicit.